### PR TITLE
test: file transfer

### PR DESCRIPTION
**Pull request by crazy-max**

### CODE REVIEW

File: test/build_test.go [hunk 1]
```diff
@@ -146,3 +148,40 @@ RUN cp /etc/foo /etc/bar
 	wg.Wait()
 	b.StopTimer()
 }
+
+func benchmarkBuildFileTransfer(b *testing.B, sb testutil.Sandbox) {
+	var appliers []fstest.Applier
+	appliers = append(appliers,
```
Comment (tonistiigi) [inline test/build_test.go:154]: This looks quite stable looking at the results so not a big issue but technically these files do not need to be recreated each time. A global helper func could just create them once and other runs can detect that they already exist. Cleanup can be after `TestMain()` in that case. Looks like it takes around 3 sec atm. I think we can push it a bit more to 5 maybe.

File: test/build_test.go [hunk 2]
```diff
@@ -146,3 +148,40 @@ RUN cp /etc/foo /etc/bar
...
+	for i := 0; i < 5000; i++ {
+		appliers = append(appliers, fstest.CreateFile(fmt.Sprintf("subdir1/file%d.txt", i+5), []byte("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."), 0600))
+	}
+	appliers = append(appliers,
+		fstest.CreateFile("subdir1/largefile1.txt", make([]byte, 1024*1024*50), 0600),  // 50MB file
+		fstest.CreateFile("subdir1/largefile2.txt", make([]byte, 1024*1024*100), 0600), // 100MB file
+	)
+
+	dockerfile := []byte(`
+FROM busybox:latest
```
Comment (tonistiigi) [inline test/build_test.go:171]: Is this `busybox` cached or pulled (or pulled from local registry). If pulled from remote then better to use local image to not skew results.

File: test/build_test.go [hunk 3]
```diff
@@ -146,3 +148,40 @@ RUN cp /etc/foo /etc/bar
...
+RUN du -sh . && tree .
+`)
+	dir := tmpdir(b,
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+		fstest.Apply(appliers...),
+	)
+
+	b.ResetTimer()
+	b.StartTimer()
+	out, err := buildxBuildCmd(sb, withArgs(dir))
```
Comment (tonistiigi) [inline test/build_test.go:183]: I think we could run this 2-3 times. Second/third run the build would be cached but the metadata context transfer would need to happen again.

File: test/build_test.go (hunk 2 above)
Comment (crazy-max) [reply test/build_test.go:171]: Yes it's cached on local registry

File: test/build_test.go (hunk 1 above)
Comment (crazy-max) [reply test/build_test.go:154]: Good point will do that

**Pull request merged**